src/pocketmine/entity/Skin.php

```diff
@@ -55,6 +55,10 @@ public function __construct(string $skinId, string $skinData, string $capeData =
 		$this->geometryData = $geometryData;
 	}
 
+	public static function convertToLegacyName(string $name) : string{
+		return '{"geometry" : {"default" : "' . $name . '"}}';
```

Comment:
This looks susceptible to injection bugs.
```suggestion
		return json_encode(["geometry" => ["default" => $name]]);
```